cric-predict/src/Components/UserGroups.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import '../Stylesheets/App.css';
+import axios from 'axios';
+import private_icon from './private.jpg';
+import {Button, Modal, Form, Row, Col}  from 'react-bootstrap';
+
+export default class UserGroupsComponent extends React.Component  {
+
+  constructor(props){
+    super(props);
+    this.state = {
+    groups: [],
+    filtered_groups: [],
+    };
+  }
+
+  async componentDidMount() {
+    const user_id = window.localStorage.getItem('user_id');
+    const response = await axios.get('http://localhost:8000/api/group/user_groups/'+user_id+"/",{
+    headers: {
+    'Authorization': "Bearer "+window.localStorage.getItem('access_token')
+    }
+})
+    const groups_list = response.data
+    console.log(groups_list)
+
+    groups_list.map((group) => (
+        group.user_count=group.users.length
+    ))
+    this.setState({groups: groups_list, filtered_groups:groups_list})
+  }
+
+    filterGroupList = async (event) => {
+            this.setState({filtered_groups:this.state.groups.filter(group => group.name.toLowerCase().includes(event.target.value.toLowerCase()))});
+        }
+
+
+  render() {
+  return (
+        <div className="card">
+            <div className="card-body">
+                <h3 className="text-style"> Your Groups</h3>
```

Comment:
we should be able to handle this in one component by aggregating data on the backend ... let me know if you need help with this